### PR TITLE
Apply scale factor on wavelength binned data.

### PR DESCRIPTION
**Pull request by YooSunYoung**
Related to #48 

We need to filter the binned data by the `hkl_eq` first, so I added `scipp.Variable` based grouping here.

### CODE REVIEW

File: src/ess/nmx/reduction.py [hunk 1]
```diff
@@ -284,42 +284,66 @@ def apply_func(val: Sequence, _cur_depth: int = 0) -> list:
 
 
 def _detour_group(
-    da: sc.DataArray, group_name: str, detour_func: Callable
+    da: sc.DataArray, group_var: str | sc.Variable, detour_func: Callable
 ) -> sc.DataArray:
     """Group the data array by a hash of a coordinate.
```
[the remark this answers is not in the document]
Comment (YooSunYoung) [reply src/ess/nmx/reduction.py:289]: @SimonHeybrock 
We need to group the data by `h, k, l` miller indices. They are stored as 3-length lists in MTZ files.

And then we need to calculate scale factor per group in the reference bin (one of the wavelength bins)

And then we need to group other bins and then apply scale factors to the intensity per group(hkl indices) per bin(wavelength).

[the remark this answers is not in the document]
Comment (YooSunYoung) [reply src/ess/nmx/reduction.py:289]: And `scipp.group` doesn't support vector3 types right...?

[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/ess/nmx/reduction.py:289]: Do we need an exact comparison of the `vector3` (is it actually integer-valued?), or a binning? Is it a 3-D grouping/binning, or 1-D?

[the remark this answers is not in the document]
Comment (YooSunYoung) [reply src/ess/nmx/reduction.py:289]: We need an exact comparison of all three indices.
It's not 3-D grouping, it'll be 1-D.

**Comment by YooSunYoung** [comment]
I'll make a new PR since the base branch was updated a lot.

**Pull request closed without merging**